file_formats/linear_format.ml

```diff
@@ -0,0 +1,100 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                    Greta Yorsh, Jane Street Europe                     *)
+(*                                                                        *)
+(*   Copyright 1996 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*   Copyright 2019 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(* marshal and unmashal a compilation unit in linear format *)
+type linear_item_info =
+  | Func of Linear.fundecl
+  | Data of Cmm.data_item list
+
+type linear_unit_info =
+  {
+    mutable unit_name : string;
+    mutable items : linear_item_info list;
+  }
+
+type error =
+  | Wrong_format of string
+  | Wrong_version of string
+  | Corrupted of string
+  | Marshal_failed of string
+
+exception Error of error
+
+let save filename linear_unit_info =
+  let ch = open_out_bin filename in
+  Misc.try_finally (fun () ->
+    output_string ch Config.linear_magic_number;
+    output_value ch linear_unit_info;
+    (* Saved because linear and emit depend on cmm.label. *)
```

Comment:
"Saved because linear and emit depend on cmm.label" -> "Saved because Linearize and Emit depend on Cmm.label"